src/refract/Element.cc

```diff
@@ -129,4 +131,243 @@ namespace refract
+             *   - with existing key - replace old value
+             * - if not member
+             *   - if empty value -> ignore (type holder for array)
+             *   - else append
+             */
+            template <typename T>
+            struct ValueMerge<T, RefractElements> {
+                typename T::ValueType& value;
+
+                ValueMerge(T& element) : value(element.value) {}
+
+                void operator()(const T& merge) {
+                    typedef std::map<std::string, MemberElement*> MapKeyToMember;
+                    MapKeyToMember keysBase;
+
+                    for (RefractElements::iterator it = value.begin() ; it != value.end() ; ++it) {
+                        if (MemberElement* member = TypeQueryVisitor::as<MemberElement>(*it)) {
+
+                            if (StringElement* key = TypeQueryVisitor::as<StringElement>(member->value.first)) {
+                                keysBase[key->value] = member;
+                            }
+                        }
+                    }
+
+                    for (RefractElements::const_iterator it = merge.value.begin() ; it != merge.value.end() ; ++it) {
+                        if (MemberElement* member = TypeQueryVisitor::as<MemberElement>(*it)) {
+                            if (StringElement* key = TypeQueryVisitor::as<StringElement>(member->value.first)) {
+                                MapKeyToMember::iterator iKey = keysBase.find(key->value);
+
+                                if (iKey != keysBase.end()) { // key is already presented, replace value
+                                    delete iKey->second->value.second;
+                                    iKey->second->value.second = member->value.second->clone();
+                                } 
+                                else { // unknown key, append value
+                                    MemberElement* clone = static_cast<MemberElement*>(member->clone());
+                                    value.push_back(clone);
+                                    keysBase[key->value] = clone;
+                                }
+                            }
+                        }
+                        else if(!(*it)->empty()) { // merge member is not MemberElement, append value
+                            value.push_back((*it)->clone());
+                        }
+                    }
+                }
+            };
+
+            template <typename T>
+            struct InfoMerge {
+                IElement::MemberElementCollection& info;
+                InfoMerge(IElement::MemberElementCollection& info) : info(info) {}
+
+                void operator()(const IElement::MemberElementCollection& append) {
+                    for (IElement::MemberElementCollection::const_iterator it = append.begin() ; it != append.end() ; ++it) {
+                        if (StringElement* key = TypeQueryVisitor::as<StringElement>((*it)->value.first)) {
```

Comment:
Is this correct? It looks like it is only overwriting existing keys but new keys are just skipped.